relay-event-schema/src/protocol/ourlog.rs

```diff
@@ -0,0 +1,253 @@
+    #[metastructure(additional_properties, retain = true, pii = "maybe", trim = false)]
+    pub other: Object<Value>,
+}
+
+#[derive(Debug, Clone, PartialEq, ProcessValue)]
+pub enum AttributeValue {
+    #[metastructure(field = "string_value", pii = "true")]
+    StringValue(String),
+    #[metastructure(field = "int_value", pii = "true")]
+    IntValue(i64),
+    #[metastructure(field = "double_value", pii = "true")]
+    DoubleValue(f64),
+    #[metastructure(field = "bool_value", pii = "true")]
+    BoolValue(bool),
+    /// Any other unknown attribute value.
+    ///
+    /// This exists to ensure other attribute values such as array and object can be added in the future.
+    Unknown(String),
+}
+
+impl IntoValue for AttributeValue {
+    fn into_value(self) -> Value {
+        let mut map = Object::new();
+        match self {
+            AttributeValue::StringValue(v) => {
+                map.insert("string_value".to_string(), Annotated::new(Value::String(v)));
+            }
+            AttributeValue::IntValue(v) => {
+                map.insert("int_value".to_string(), Annotated::new(Value::I64(v)));
+            }
+            AttributeValue::DoubleValue(v) => {
+                map.insert("double_value".to_string(), Annotated::new(Value::F64(v)));
+            }
+            AttributeValue::BoolValue(v) => {
+                map.insert("bool_value".to_string(), Annotated::new(Value::Bool(v)));
+            }
+            AttributeValue::Unknown(v) => {
+                map.insert("unknown".to_string(), Annotated::new(Value::String(v)));
+            }
+        }
+        Value::Object(map)
+    }
+
+    fn serialize_payload<S>(&self, s: S, _behavior: SkipSerialization) -> Result<S::Ok, S::Error>
+    where
+        Self: Sized,
+        S: serde::Serializer,
+    {
+        let mut map = s.serialize_map(None)?;
+        match self {
+            AttributeValue::StringValue(v) => {
+                map.serialize_entry("string_value", v)?;
+            }
+            AttributeValue::IntValue(v) => {
+                map.serialize_entry("int_value", v)?;
+            }
+            AttributeValue::DoubleValue(v) => {
+                map.serialize_entry("double_value", v)?;
+            }
+            AttributeValue::BoolValue(v) => {
+                map.serialize_entry("bool_value", v)?;
+            }
+            AttributeValue::Unknown(v) => {
+                map.serialize_entry("unknown", v)?;
+            }
+        }
+        map.end()
+    }
+}
+
+impl AttributeValue {
+    /// Returns the string representation of this attribute value.
+    pub fn as_str(&self) -> String {
+        match self {
+            AttributeValue::StringValue(s) => s.clone(),
+            AttributeValue::IntValue(s) => s.to_string(),
+            AttributeValue::DoubleValue(s) => s.to_string(),
+            AttributeValue::BoolValue(s) => s.to_string(),
+            AttributeValue::Unknown(s) => s.clone(),
+        }
+    }
```

Comment:
Is this function used anywhere?

nit: `as_str` usually returns `&str`, `to_string` returns `String`.